basic-examples/quoting/README.md

```diff
@@ -0,0 +1,48 @@
+# Quoting
+
+## Workflow Description
+
+This workflow demonstrates example expressions with quotes as detailed in our [expression documentation](https://arcalot.io/arcaflow/workflows/expressions/).
+
+There are multiple types of quoting supported. Quotes must pass through YAML's syntax; both the expression language and YAML itself process the quotes and escaped characters, and the input must be valid for both. Consider using YAML "Block Flow Scalar" format to prevent YAML from changing your quotes, and using Arcaflow expression "raw" strings (with back-ticks `) to not need to escape the quotes. In addition, you can use single or double quotes.

```

Comment:
Not really sure I agree that block flow + arca "raw" is "obviously" the best, since raw strings suppress interpretation of embedded controls like `\n` as well as allowing nested identical quotes. Sometimes that's what you want; and, yeah, if all you want is to quote `"foo 'bar'"`, raw strings are a convenient short cut, but it's not very general.

(Adding something like a `\t` to the example instead of just nested quotes would clarify the tradeoffs.)

Reply:
Yeah, I think that the right answer is that there is no right answer, here.  The challenge is that things have to be quoted once for YAML and once for ARCA, and that we only have three kinds of quotation marks, so things get limited very fast if you want to embed one or more of the three into your string literal.

Block flow scalar format helps with that, because then YAML doesn't require using up one of the marks.  Raw strings are nice because then we don't have to worry about any marks other than `` ` ``.  But, as you say, if you _do_ want a special character in the string (such as a horizontal tab) and you don't want to put a literal one in there, then raw strings are not your friend...and, you need to cozy up to escapes.  (But, then you also, potentially, have to worry about escaping them from both YAML and ARCA, and then the backslashes start piling up....  🙃)

Reply:
This goes back to the "who the heck ever thought using YAML for a programming language was a good idea?" ... double quoting is never a friend. That's why raw quotes exist, because it's a pain for Python and Go and any other REGEX language where quoted special characters are a fact of life... but when the raw quoted strings need to go through another layer of parsing, the complication is magnified. 😦

Reply:
Yeah, but the block-flow thing is pretty close to "raw", and then the backquotes are handed to ARCA which treats them as marking a "raw" string, and so the combination is pretty uncooked all the way through.

But, yeah, stepping back from that is walking into a minefield, given that the user becomes a servant to two masters.

Reply:
I still think this just grabs the reader's hand and pulls them straight into the middle of the minefield without a map; but I approved, so, "whatever". I'll resolve since no correction is apparently forthcoming. 😆

Reply:
As proposed, it just says "consider", which I think is the right way to go. I don't think it's an obvious option, it's just an option.

Reply:
Yeah...the thing is, simple strings are simple (as they should be).  Things get tricker when you want tabs and newlines in there, and for that you probably want "normal" strings with escape characters.  But, then, if you want to put double-quotes or apostrophes into strings, then (_**I**_ think) raw strings are a better way to go...right up until you want _both_ things.

What we should be promoting depends on what the user wants to do.  Nevertheless, since this example is principally about embedding quotes in strings, I think highlighting YAML block flow scalars and ARCA raw strings is an excellent idea, since these are likely to be unfamiliar concepts.  (Block flow scalars are good in lots of instances, if you can tolerate the extra line; whether to use raw strings or not really depends on what is in the strings.)